RecoLocalCalo/EcalRecProducers/plugins/Common.h

```diff
@@ -0,0 +1,46 @@
+#ifndef RecoLocalCalo_EcalRecProducers_plugins_Common_h
+#define RecoLocalCalo_EcalRecProducers_plugins_Common_h
+
+#include <cassert>
+#include <chrono>
+#include <cmath>
+#include <cstdint>
+#include <iostream>
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+// a workaround for std::abs not being a constexpr function
+namespace ecal {
+
+  template <typename T>
+  constexpr T abs(T const& value) {
+    return ::std::max(value, -value);
+  }
```

Comment:
According to the [CUDA documentation](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH.html#group__CUDA__MATH) and to a simple test I've tried, this should not be needed (any more ?).
@thomreis could you try removing it and replacing any instance with `std::abs()` ?

Reply:
Done in https://github.com/cms-patatrack/cmssw/pull/591/commits/7060f0cbae7027baac8bd0be66e030be6e033cee